diff --git a/src/i18n.ts b/src/i18n.ts
index 2be7ea0..2374842 100644
--- a/src/i18n.ts
+++ b/src/i18n.ts
@@ -60,7 +60,9 @@ export const setLanguage = async (newLng: string | undefined) => {
   currentLanguage =
     languages.find((language) => language.lng === newLng) || fallbackLanguage;
 
-  document.documentElement.dir = currentLanguage.rtl ? "rtl" : "ltr";
+  try {
+    document.documentElement.dir = currentLanguage.rtl ? "rtl" : "ltr";
+  } catch (e) {}
 
   currentLanguageData = await import(`./locales/${currentLanguage.data}`);
 
diff --git a/src/keys.ts b/src/keys.ts
index 07cce39..4f652da 100644
--- a/src/keys.ts
+++ b/src/keys.ts
@@ -1,4 +1,6 @@
-export const isDarwin = /Mac|iPod|iPhone|iPad/.test(window.navigator.platform);
+export const isDarwin = /Mac|iPod|iPhone|iPad/.test(
+  typeof window !== "undefined" ? window.navigator.platform : "",
+);
 
 export const KEYS = {
   ARROW_LEFT: "ArrowLeft",
